src/main/java/com/cozyhome/onlineshop/handler/CozyHomeExceptionHandler.java

```diff
@@ -24,59 +27,75 @@
 import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
-import java.time.LocalDateTime;
+import java.util.List;
+import java.util.NoSuchElementException;
 
 @Slf4j(topic = "COZY_HOME_EXCEPTION_HANDLER")
 @RestControllerAdvice
 public class CozyHomeExceptionHandler extends ResponseEntityExceptionHandler {
 
-    @ExceptionHandler({DataNotFoundException.class, IllegalArgumentException.class})
+    @Value("${reflectoring.trace}")
+    private boolean printStackTrace;
+
+    @ExceptionHandler({DataNotFoundException.class, IllegalArgumentException.class, NoSuchElementException.class})
```

Comment:
Do we have any difference in use between DataNotFoundException, IllegalArgumentEx. and NoSuchElementEx.?

Reply:
we need to write all the errors in ExceptionHandler we can get

Reply:
My point is that we should have a single strategy of using DataNotFoundException or IllegalArgumentException